src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsRepository.cs

```diff
@@ -173,11 +173,22 @@ private string GetSecretsBlobPath(ScriptSecretsType secretsType, string function
 
         private async Task WriteToBlobAsync(string blobPath, string secretsContent)
         {
-            BlockBlobClient secretBlobClient = Container.GetBlockBlobClient(blobPath);
-            using (StreamWriter writer = new StreamWriter(await secretBlobClient.OpenWriteAsync(true)))
+            BlobClient secretBlobClient = Container.GetBlobClient(blobPath);
+            BlobUploadOptions uploadOptions = new BlobUploadOptions();
+
+            if (await secretBlobClient.ExistsAsync())
+            {
+                // Return a 412 if another write beats us to updating the file.
```

Comment:
It would be good to track work to detect those conditions and better handle the conflict scenarios. This is a huge improvement, but will still cause an exception that will bubble up here and impact an instance of the host, or at least an API call. 

That enhancement would be outside of the scope of this PR, but it would be great if we handle the conflicts and cause the manager to load the new secrets instead of throwing. Would you mind opening an issue to discuss?

Reply:
Good point. I filed https://github.com/Azure/azure-functions-host/issues/8731 to track.